### PR TITLE
Bid & BlindBid service basic implementations

### LINKED ISSUE #2: Create the basis for the bid service
We need to create the bid service and define all of the structures methods & messages that it would require. 

There are some things that are still not clear such as the signature field for the `ExtendBid` function, but the rest is quite clear and therefore we should start working on it in order to be able to work on the `Bid` service integration in `rusk`.

### PR DESCRIPTION
**Pull request by CPerezz**
Closes #2

### CODE REVIEW

File: bid.proto [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import "basic_fields.proto";
+import "transaction.proto";
+import "keys.proto";
+
+message Bid {
+    PoseidonCypher encrypted_data = 1;
+    BlsScalar nonce = 2;
+    BlsScalar hashed_secret_k = 3;
+    JubJubCompressed R = 4;
+    JubJubCompressed pk = 5;
```
Comment (vlopes11) [inline bid.proto]: Shouldn't be pk_r instead?

File: bid.proto [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    JubJubCompressed c = 6;
+}
+
+// We will need to send a secret to encrypt the value, blinder
+// & secret
+message BidTransactionRequest {
+    // TODO: Review type
+    bytes K = 1;
+    BlsScalar bid_value = 2;
+    JubJubCompressed pk = 3;
```
Comment (vlopes11) [inline bid.proto]: Same as above

File: bid.proto [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    BlsScalar seed = 5;
+    fixed64 expiration_ts = 6;
+    uint32 elegibility_ts = 7;
+    // The call will edit the Tx `calldata` field with the Bid fn call
+    Transaction tx = 8;
+}
+
+message BidTransactionResponse {
+    Bid bid = 1;
+    BlsScalar bid_hash = 2;
```
Comment (vlopes11) [inline bid.proto]: The bid hash is part of the response because we assume the receiver won't be able to calculate the hash of the bid? Considering the hash is calculated only with public data.

File: blindbid.proto [hunk 4]
```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+package rusk;
+
+import "basic_fields.proto";
+import "transaction.proto";
+import "keys.proto";
+
+message GenerateScoreRequest {
```
Comment (vlopes11) [inline blindbid.proto:15]: `secret_k` and `tree_root` should be also provided

File: blindbid.proto [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    Proof blindbid_proof = 1;
+    BlsScalar score = 2;
+    BlsScalar prover_identity = 4;
+}
+
+message VerifyScoreRequest {
+    Proof proof = 1;
+    BlsScalar score = 2;
+    BlsScalar seed = 3;
+    BlsScalar identity = 4;
```
Comment (vlopes11) [inline blindbid.proto]: Is `identity` the `prover_id`?

File: transaction.proto [hunk 6]
```diff
@@ -0,0 +1,40 @@
+syntax = "proto3"
```
Comment (vlopes11) [inline transaction.proto:1]: How is this related to the blind bid?

**Review by vlopes11 — CHANGES_REQUESTED** [review]
A few questions

File: bid.proto (hunk 3 above)
Comment (CPerezz) [reply bid.proto]: He will not be able ye. Since go has no impl for sponge hash.

File: blindbid.proto (hunk 4 above)
Comment (CPerezz) [reply blindbid.proto:15]: About bid_tree_root is obtained in rusk's side. Since it has the storage, it can check the root of the tree. Go has no way to know it. Therefore, it should not be added in the request.

Secret_k makes sense, I forgot. Will add it

File: blindbid.proto (hunk 5 above)
Comment (CPerezz) [reply blindbid.proto]: Yes. Will make it more explicit

File: transaction.proto (hunk 6 above)
Comment (CPerezz) [reply transaction.proto:1]: It's not. But I needed the types in order to be able to write all of the service messages.

File: bid.proto (hunk 2 above)
Comment (CPerezz) [reply bid.proto]: Done

File: bid.proto (hunk 1 above)
Comment (CPerezz) [reply bid.proto]: Done

File: blindbid.proto [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+service BlindBid {
+    // Generate a proof of blind bid, and a score.
+    rpc GenerateScore(GenerateScoreRequest) returns (GenerateScoreResponse) {}
+    // Verify a proof of blind bid, and the associated score.
+    rpc VerifyScore(VerifyScoreRequest) returns (VerifyScoreResponse) {}
+}
+
+message GenerateScoreRequest {
+    BlsScalar secret_k + 1;
```
Comment (vlopes11) [inline blindbid.proto]: Seems to have a typo here

**Review by vlopes11 — CHANGES_REQUESTED** [review]

File: blindbid.proto (hunk 7 above)
Comment (CPerezz) [reply blindbid.proto]: Ty

**Review by vlopes11 — APPROVED** [review]
LGTM

**Comment by autholykos** [comment]
LGTM

**Comment by CPerezz** [comment]
@ZER0 once you're back It would be nice for you to review this also!
